compiler/rustc_metadata/src/rmeta/decoder/cstore_impl.rs

```diff
@@ -226,7 +226,14 @@ provide! { tcx, def_id, other, cdata,
     lookup_default_body_stability => { table }
     lookup_deprecation_entry => { table }
     params_in_repr => { table }
-    unused_generic_params => { table }
+    unused_generic_params => {
```

Comment:
You had a FIXME in the other PR, could you add this here as well? Would be nice to just default

Reply:
Oh also, is there a reason why the type doesn't implement that trait? Could you just add the impl? Or is there something to it

Reply:
For us to have a defaulted table, the value must implement `FixedSizeEncoding` and `IsDefault`. `Lazy<T>` is `FixedSizeEncoding`, but can't be checked for defaultness because they're just a file offset.

I could instead encode the table value as `Option<LazyValue<UnusedGenericParams>>`, but then `ProcessQueryValue<T> for Option<T>` panics if the value decoded is `None` -- we *could* use specializtion to do:

```rust
impl ProcessQueryValue<'_, UnusedGenericParams> for Option<UnusedGenericParams> {
    #[inline(always)]
    fn process_decoded(self, _tcx: TyCtxt<'_>, _err: impl Fn() -> !) -> UnusedGenericParams {
        self.unwrap_or_else(|| UnusedGenericParams::new_all_used())
    }
}
```

but I don't feel like using specialization here is compelling enough to save a few lines in the decoder rather than providing an explicit body here.

Reply:
haha, yeah, it's fine